torchtext/data/functional.py

```diff
@@ -114,3 +115,72 @@ def _internal_func(txt_iter):
         for line in txt_iter:
             yield sp_model.EncodeAsPieces(line)
     return _internal_func
+
+
+def compute_ngrams(tokens, n):
+    """ Compute n-grams for a list of tokens
+
+    Arguments:
+        tokens: a list of tokens (typically a string split on whitespaces)
+        n: the order of n-gram wanted
+
+    Outputs:
+        output: a list of tuples with the n-grams from the input list of tokens
+
+    Examples:
+        >>> from torchtext.data.functional import compute_ngrams
+        >>> tokens = ['This', 'is', 'definitely', 'me']
+        >>> compute_ngrams(tokens, 2)
+            [('This', 'is'), ('is', 'definitely'), ('definitely',
+            'me')]]
+    """
+    return list(zip(*[tokens[j:] for j in range(n)]))
+
+
+def compute_ngram_counter(tokens, max_n):
```

Comment:
And all the functions need a docs and unit test so it's better to minimize the number of functions.